core/src/test/scala/org/apache/spark/scheduler/CacheRecoveryIntegrationSuite.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import scala.util.Try
+
+import org.scalatest.{BeforeAndAfterEach, Matchers}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.time.{Seconds, Span}
+
+import org.apache.spark.{SparkConf, SparkContext, SparkFunSuite, TestUtils}
+import org.apache.spark.internal.config._
+import org.apache.spark.network.TransportContext
+import org.apache.spark.network.netty.SparkTransportConf
+import org.apache.spark.network.server.TransportServer
+import org.apache.spark.network.shuffle.ExternalShuffleBlockHandler
+import org.apache.spark.rdd.RDD
+import org.apache.spark.storage._
+
+class CacheRecoveryIntegrationSuite
```

Comment:
I'm not too sure about what this suite is testing. You seem to be relying on the very short idle timeout to have the executors killed; but it doesn't seem like there a strong relation between your test code and which / how many executors will be killed as idle.

Instead, shouldn't you explicitly be killing executors to verify that the data is being replicated?

Or at least explain in a comment how this test is actually testing the functionality being added.

Reply:
This class is meant to be an integration test that shows cache recovery working in a real cluster, unlike the unit tests in CacheRecoveryManagerSuite, where much of the cluster machinery is mocked. That is why i do not shut down executors directly and why I have to inspect the results in a roundabout way. I added some comments explaining this. 

If you don't think this kind of testing is necessary, we can take it out, but I thought it was a nice sanity check while developing.